lib/sequenceserver/doctor.rb

```diff
@@ -0,0 +1,185 @@
+      # access permissions.
+      def inspect_file_access
+        File.readable?(config.data[:database_dir]) &&
+          File.writable?(config.data[:database_dir]) &&
+          File.writable?(config.config_file)
+      end
+
+      # Retrieve sequence ids (specified by %i) from all databases. Using
+      # accession number is problematic because of several reasons.[2]
+      def all_sequence_ids(ignore)
+        Database.map do |db|
+          next if ignore.include? db
+
+          out = `blastdbcmd -entry all -db #{db.name} -outfmt "%i" 2> /dev/null`
+          {
+            :db     => db,
+            :seqids => out.to_s.split
+          }
+        end.compact
+      end
+
+      # FASTA files formatted without -parse_seqids option won't support the
+      # blastdbcmd command of fetching sequence ids using '%i' identifier.
+      # In such cases, an array of 'N/A' values are returned which is
+      # checked in this case.
+      def inspect_parse_seqids(seqids)
+        seqids.map do |sq|
+          sq[:db] if sq[:seqids].include? 'N/A'
+        end.compact
+      end
+
+      # Pretty print database list.
+      def bullet_list(values)
+        list = ''
+        values.each do |value|
+          list << "      - #{value}\n"
+        end
+        list
+      end
+
+      # Print diagnostic error messages according to the type of error.
+      # rubocop:disable Metrics/MethodLength
+      def show_message(error, values)
+        return if values.empty?
+
+        case error
+        when ERROR_PARSEQ_IDS
+          puts <<MSG
+*** Doctor has found improperly formatted database:
+#{bullet_list(values)}
+    Please reformat your databases with -parse_seqids switch (or use
+    sequenceserver -m) for using SequenceServer, as the current one
+    is likely to cause problems.
+MSG
+        when ERROR_NUMERIC_IDS
+          puts <<MSG
+*** Doctor has found databases with numeric sequence ids:
+#{bullet_list(values)}
+    Note that this could likely cause problems with sequence retrieval.
+MSG
+
+        when ERROR_PROBLEMATIC_IDS
+          puts <<MSG
+*** Doctor has found databases with problematic sequence ids:
+#{bullet_list(values)}
+    This causes some sequence to contain extraneous words like `gnl|`
+    appended to their id string.
+MSG
+
+        when ERROR_BAD_FILE_PERMISSION
+          puts <<MSG
+*** Doctor has found bad file permission settings for one of the required files.
+#{bullet_list(values)}
+    Please ensure that config file and databases have appropriate read/write
+    permissions.
+MSG
+        end
+      end
+      # rubocop:disable Metrics/MethodLength
+    end
+
+    def initialize
+      @ignore     = []
+      @all_seqids = Doctor.all_sequence_ids(@ignore)
+    end
+
+    attr_reader :invalids, :all_seqids
+
+    def diagnose
+      puts "\n1/4 Inspecting databases for proper -parseq_ids formatting.."
+
+      check_parse_seqids
+
+      remove_invalid_databases
```

Comment:
You should probably have, at some level, some output that you're deleting things here.